### PR TITLE
Upgrade to sam2

### PR DESCRIPTION
**Pull request by harshbhatt7585**

### CODE REVIEW

**Comment by salma-remyx** [comment]
LGTM 👍 We can update the SAM2 install later if the transformers library includes it

File: docker/segment_processor/process_segment.py [hunk 1]
```diff
@@ -4,26 +4,65 @@
...
+    labels = parsed_answer['labels']
+    return list(zip(bboxes, labels))
+
 
 def segment_image_data(row):
     try:
-        preds = clipseg.run_inference(row["image"], row["captions"])
 
-        sampled_points = []
+        caption_points = florence_caption(row["image"])
```
Comment (salma-remyx) [inline docker/segment_processor/process_segment.py:58]: @harshbhatt7585 this returns bounding boxes so we may need to use another method of SAM2 accepting bbox prompts

Comment (harshbhatt7585) [reply docker/segment_processor/process_segment.py:58]: yes, I yet have to test this, I will fix this

**Comment by salma-remyx** [comment]
addressed by #14

**Pull request closed without merging**